bug(tests): cocoindex @target_connector registry collision under pytest-xdist

## Summary

Two tests in `tests/unit/ingestion/test_target_sync_execution.py` fail intermittently when running the full unit suite with `pytest-xdist` (`-n auto`). They **always pass** when run in isolation.

## Affected Tests

```
FAILED tests/unit/ingestion/test_target_sync_execution.py::TestTargetSyncExecution::test_mutate_does_not_call_asyncio_run
FAILED tests/unit/ingestion/test_target_sync_execution.py::TestTargetSyncExecution::test_handle_methods_are_sync
```

(3rd test `test_writer_has_sync_methods` doesn't import the target module, so it's unaffected.)

## Full Traceback

```
tests/unit/ingestion/test_target_sync_execution.py:23: in test_handle_methods_are_sync
    from src.ingestion.unified.targets.qdrant_hybrid_target import (
src/ingestion/unified/targets/qdrant_hybrid_target.py:102: in <module>
    @target_connector(
.venv/lib/python3.12/site-packages/cocoindex/op.py:1092: in _inner
    _engine.register_target_connector(spec_cls.__name__, connector)
E   RuntimeError: Target factory with name already exists: QdrantHybridTargetSpec
```

Both tests fail with the same error — they both `from src.ingestion.unified.targets.qdrant_hybrid_target import QdrantHybridTargetConnector`.

## Root Cause

The `@target_connector` decorator in cocoindex (v0.3.31) registers the target spec in a **global C++ engine registry** (`_engine.register_target_connector`). When `pytest-xdist` assigns multiple test modules that import `qdrant_hybrid_target.py` to the **same worker process**, the module gets re-imported and the `@target_connector` decorator fires a second time, causing:

```
RuntimeError: Target factory with name already exists: QdrantHybridTargetSpec
```

This is an xdist worker-level collision — the decorator is idempotent within a single import but not across xdist's `forked_run_report` reimporting.

**Passes in isolation** because only one import path exists per process.

## Reproduction

```bash
# FAILS (intermittently — depends on xdist worker assignment)
uv run pytest tests/unit/ -n auto --timeout=30

# PASSES (always)
uv run pytest tests/unit/ingestion/test_target_sync_execution.py -v
uv run pytest tests/unit/ingestion/test_target_sync_execution.py -v -n auto
```

## Possible Fixes

1. **Guard the import** in tests — use `importlib` to check if already registered before importing:
   ```python
   @pytest.fixture(autouse=True)
   def _skip_if_already_registered():
       try:
           from src.ingestion.unified.targets.qdrant_hybrid_target import QdrantHybridTargetConnector
       except RuntimeError:
           pytest.skip("cocoindex registry collision under xdist")
   ```

2. **Isolate tests via xdist group** — force these tests onto their own worker:
   ```python
   @pytest.mark.xdist_group("cocoindex")
   class TestTargetSyncExecution: ...
   ```

3. **Lazy import guard in the module itself** — wrap `@target_connector` with a registration check (requires cocoindex API to support `is_registered()`).

4. **Upstream fix** — report to cocoindex: `register_target_connector` should be idempotent (skip if name already exists).

## Environment

- Python 3.12.3
- cocoindex 0.3.31
- pytest-xdist 3.8.0
- `--dist loadscope` / `--dist worksteal`

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -380,6 +380,7 @@ markers = [
     "slow: Tests taking > 5 seconds",
     "legacy_api: tests for pre-LangGraph API (may need rewrite)",
     "smoke: Smoke tests (requires running services)",
+    "xdist_group: Group tests onto same xdist worker (prevents global registry collisions)",
 ]
 
 # =============================================================================

--- a/tests/unit/ingestion/test_target_sync_execution.py
+++ b/tests/unit/ingestion/test_target_sync_execution.py
@@ -3,9 +3,26 @@
 
 import asyncio
 
+import pytest
 
+
+@pytest.mark.xdist_group("cocoindex")
 class TestTargetSyncExecution:
     """Test that mutate() works without asyncio.run() conflicts."""
+
+    @pytest.fixture(autouse=True)
+    def _skip_if_registry_collision(self):
+        """Skip gracefully if cocoindex target_connector is already registered."""
+        try:
+            from src.ingestion.unified.targets.qdrant_hybrid_target import (  # noqa: F401
+                QdrantHybridTargetConnector,
+            )
+        except ImportError as exc:
+            pytest.skip(f"cocoindex not installed: {exc}")
+        except RuntimeError as exc:
+            if "already exists" in str(exc):
+                pytest.skip(f"cocoindex registry collision under xdist: {exc}")
+            raise
 
     def test_mutate_does_not_call_asyncio_run(self):
         """mutate() should not use asyncio.run() directly."""

